src/lib/util/corestr.cpp

```diff
@@ -171,27 +171,41 @@ void strreplacechr(std::string& str, char ch, char newch)
 	}
 }
 
-std::string strtrimspace(std::string& str)
+static std::string internal_strtrimspace(std::string& str, bool right_only)
 {
-	int start = 0;
-	for (auto & elem : str)
-	{
-		if (!isspace(uint8_t(elem)))  break;
-		start++;
-	}
-	int end = str.length();
-	if (end > 0)
+	// identify the start
+	std::string::iterator start = str.begin();
+	if (!right_only)
 	{
-		for (size_t i = str.length() - 1; i > 0; i--)
-		{
-			if (!isspace(uint8_t(str[i]))) break;
-			end--;
-		}
+		start = std::find_if(
+			str.begin(),
+			str.end(),
+			[](char c) { return !isspace(c); });
```

Comment:
This removes the cast from `char` to `uint8_t` before calling `isspace`.  This will cause a change in behaviour on platforms where `char` is signed (e.g. i386 and x86-64) when the top bit of `c` is set.